app/src/main/java/com/example/playlistmaker/SettingsActivity.kt

```diff
@@ -12,10 +13,39 @@ class SettingsActivity : AppCompatActivity() {
         setContentView(R.layout.activity_settings)
 
         val backButton = findViewById<ImageButton>(R.id.back)
+        val shareButton = findViewById<Button>(R.id.theme_switcher2)
+        val supportButton = findViewById<Button>(R.id.theme_switcher3)
+        val termsButton = findViewById<Button>(R.id.theme_switcher4)
+
+        // Возвращаемся назад без создания нового MainActivity
         backButton.setOnClickListener {
-            val intent = Intent(this, MainActivity::class.java)
-            startActivity(intent)
             finish()
         }
+
+        // Кнопка "Поделиться приложением"
+        shareButton.setOnClickListener {
+            val shareIntent = Intent(Intent.ACTION_SEND).apply {
+                type = "text/plain"
+                putExtra(Intent.EXTRA_TEXT, getString(R.string.share_text))
+            }
+            startActivity(Intent.createChooser(shareIntent, getString(R.string.share)))
+        }
+
+        // Кнопка "Написать в поддержку"
+        supportButton.setOnClickListener {
+            val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
+                data = Uri.parse("mailto:${getString(R.string.support_email)}")
+                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
+                putExtra(Intent.EXTRA_TEXT, getString(R.string.email_body))
+            }
```

Comment:
Для корректного заполнения полей необходимо настроить интент следующим образом:
```suggestion
        val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
                data = "mailto:".toUri()
                putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.support_email)))
                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
                putExtra(Intent.EXTRA_TEXT, getString(R.string.email_body))
            }
```